Data/IntSet/Internal.hs

```diff
@@ -661,102 +668,101 @@ isSubsetOf Nil _         = True
 --------------------------------------------------------------------}
 -- | /O(n)/. Filter all elements that satisfy some predicate.
 filter :: (Key -> Bool) -> IntSet -> IntSet
-filter predicate t
-  = case t of
-      Bin p m l r
-        -> bin p m (filter predicate l) (filter predicate r)
-      Tip kx bm
-        -> tip kx (foldl'Bits 0 (bitPred kx) 0 bm)
-      Nil -> Nil
-  where bitPred kx bm bi | predicate (kx + bi) = bm .|. bitmapOfSuffix bi
-                         | otherwise           = bm
-        {-# INLINE bitPred #-}
+filter predicate t = case t of
+  Bin p m l r -> bin p m (filter predicate l) (filter predicate r)
+  Tip kx bm   -> tip kx (foldl'Bits 0 (bitPred kx) 0 bm)
+  Nil         -> Nil
+  where
+    bitPred kx bm bi | predicate (kx + bi) = bm .|. bitmapOfSuffix bi
+                     | otherwise = bm
+    {-# INLINE bitPred #-}
 
--- | /O(n)/. partition the set according to some predicate.
-partition :: (Key -> Bool) -> IntSet -> (IntSet,IntSet)
+partition :: (Key -> Bool) -> IntSet -> (IntSet, IntSet)
 partition predicate0 t0 = toPair $ go predicate0 t0
   where
-    go predicate t
-      = case t of
-          Bin p m l r
-            -> let (l1 :*: l2) = go predicate l
-                   (r1 :*: r2) = go predicate r
-               in bin p m l1 r1 :*: bin p m l2 r2
-          Tip kx bm
-            -> let bm1 = foldl'Bits 0 (bitPred kx) 0 bm
-               in  tip kx bm1 :*: tip kx (bm `xor` bm1)
-          Nil -> (Nil :*: Nil)
-      where bitPred kx bm bi | predicate (kx + bi) = bm .|. bitmapOfSuffix bi
-                             | otherwise           = bm
-            {-# INLINE bitPred #-}
+    go predicate t = case t of
+      Bin p m l r ->
+        let (l1:*:l2) = go predicate l
+            (r1:*:r2) = go predicate r
+        in  bin p m l1 r1 :*: bin p m l2 r2
+      Tip kx bm ->
+        let bm1 = foldl'Bits 0 (bitPred kx) 0 bm
+        in  tip kx bm1 :*: tip kx (bm `xor` bm1)
+      Nil -> (Nil :*: Nil)
+      where
+        bitPred kx bm bi | predicate (kx + bi) = bm .|. bitmapOfSuffix bi
+                         | otherwise = bm
+        {-# INLINE bitPred #-}
+
 
 
 -- | /O(min(n,W))/. The expression (@'split' x set@) is a pair @(set1,set2)@
 -- where @set1@ comprises the elements of @set@ less than @x@ and @set2@
 -- comprises the elements of @set@ greater than @x@.
 --
 -- > split 3 (fromList [1..5]) == (fromList [1,2], fromList [4,5])
-split :: Key -> IntSet -> (IntSet,IntSet)
-split x t =
-  case t of
-      Bin _ m l r
-          | m < 0 -> if x >= 0  -- handle negative numbers.
-                     then case go x l of (lt :*: gt) -> let !lt' = union lt r
-                                                        in (lt', gt)
-                     else case go x r of (lt :*: gt) -> let !gt' = union gt l
-                                                        in (lt, gt')
-      _ -> case go x t of
-          (lt :*: gt) -> (lt, gt)
+split :: Key -> IntSet -> (IntSet, IntSet)
```

Comment:
This doesn't strike me as very readable. What do you think?

Reply:
Yeah, I'm not a huge fan of how it formatted this one. I think having the `case`, `let` on the same line helps readability. I'll revert this change.